File: Day64/maxPAthSum.js
class Solution {
    public:
      //Function to return maximum path sum from any node in a tree.
      
      int fsh(Node* root,int& res)
      {
          if(!root)return 0;
          
          int l = max(0,fsh(root->left,res));
          int r = max(0,fsh(root->right,res));
          
          res=max(res,root->data+l+r);
          
          return max(l+root->data,r+root->data);
          
      }
      
      
      int findMaxSum(Node* root)
      {
          int res = INT_MIN;
          fsh(root,res);
          return res;
      }
  };

File: Day206/maxPAthSum.cpp
class Solution {
  public:
    int max;
    //Function to return maximum path sum from any node in a tree.
    int findMaxSum(Node* root)
    {
        max = INT_MIN;
        findMaxSum1(root);
        return max;
    }
    
    int findMaxSum1(Node* root)
    {
        if(!root)return 0;
        
        int l = std::max(0,findMaxSum1(root->left));
        int r = std::max(0,findMaxSum1(root->right));
        
        max = std::max(max,l+r+root->data);
        
        return std::max(l,r) + root->data;
    }
};
